### ISSUE TITLE
TypeVar causes name error on class-scoped variable

### ISSUE THREAD

**Issue by dmtucker** [python/mypy#9578]
**Bug Report**

Defining a `TypeVar` causes a false-positive "Name is not defined" error on class-scoped variables.

**To Reproduce**

``` python
from typing import List, TypeVar
import attr

_Foo = TypeVar('_Foo', bound='Foo')  # comment this line and the problem goes away

@attr.s
class Foo:
    _type = List[str]
    bar = attr.ib(type=_type)
```

**Expected Behavior**

```
Success: no issues found in 1 source file
```

**Actual Behavior**

```
sscce.py:9: error: Name '_type' is not defined
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: `mypy 0.790`
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: `Python 3.8.5`
- Operating system and version: `Pop!_OS 20.04 LTS`

**Comment by erictraut** [comment]
This appears to have been fixed. I can't repro it with the latest version of mypy.

**Comment by dmtucker** [comment]
Indeed! Appears to have been fixed in 0.961

FF: 0.960 emits an `AssertionError` (with `attrs-23.1.0`)
```
Traceback (most recent call last):
  File "/tmp/tmp.AyxcqWB2Wv/venv/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/tmp/tmp.AyxcqWB2Wv/venv/lib/python3.10/site-packages/mypy/__main__.py", line 12, in console_entry
    main(None, sys.stdout, sys.stderr)
  File "mypy/main.py", line 96, in main
  File "mypy/main.py", line 173, in run_build
  File "mypy/build.py", line 154, in build
  File "mypy/build.py", line 230, in _build
  File "mypy/build.py", line 2729, in dispatch
  File "mypy/build.py", line 3087, in process_graph
  File "mypy/build.py", line 3179, in process_stale_scc
  File "mypy/semanal_main.py", line 88, in semantic_analysis_for_scc
  File "mypy/semanal_main.py", line 414, in apply_class_plugin_hooks
  File "mypy/semanal_main.py", line 434, in apply_hooks_to_class
  File "mypy/plugins/attrs.py", line 309, in attr_class_maker_callback
  File "mypy/plugins/attrs.py", line 372, in _analyze_class
  File "mypy/plugins/attrs.py", line 498, in _attributes_from_assignment
  File "mypy/plugins/attrs.py", line 594, in _attribute_from_attrib_maker
  File "mypy/semanal.py", line 5450, in anal_type
  File "mypy/types.py", line 765, in accept
  File "mypy/typeanal.py", line 172, in visit_unbound_type
  File "mypy/typeanal.py", line 180, in visit_unbound_type_nonoptional
  File "mypy/semanal.py", line 4563, in lookup_qualified
  File "mypy/semanal.py", line 4458, in lookup
  File "mypy/semanal.py", line 4508, in is_active_symbol_in_class_body
AssertionError
```
